pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1537,17 +1537,31 @@ func buildListener(opts buildListenerOpts, trafficDirection core.TrafficDirectio
 	switch opts.transport {
 	case istionetworking.TransportProtocolTCP:
 		var bindToPort *wrappers.BoolValue
+		var connectionBalance *listener.Listener_ConnectionBalanceConfig
 		if !opts.bindToPort {
 			bindToPort = proto.BoolFalse
 		}
+
+		if opts.proxy.Metadata.OutboundListenerExactBalance && trafficDirection == core.TrafficDirection_OUTBOUND {
+			// virtualOutbound listener should not use exact_balance as it is using
+			// original dst and the associated redirected listeners are not bound
+			// to a port, per envoy docs
+			connectionBalance = &listener.Listener_ConnectionBalanceConfig{
```

Comment:
Do not understand comment at this place - are you saying since this is not VirtualOutbound you are adding the exact balance? It is little confusing to me.